calcom/cal.com
# feat: existing typos, add `cspell` in pre-commit to avoid future typos

## What does this PR do?



Current version of the code have typos which counts to around 120 unique word. I feel a well maintained project should not have typos. So, I have fixed them. 

Also to void future typos, I have added a config setup in pre-commit hook using the [cspell](https://www.npmjs.com/package/cspell) package. This will prevent users from committing code with unintentional mis spellings. 

Created a [file](https://github.com/kavinkuma6/cal.com/blob/add-cspell-pre-commit-to-fix-typos/.cspell/projectWords.txt), which contains the project specific words or words not found dictionary. This can be used by the package. In future, incase if the user wants to add a non dictionary word to the code,  the same should be added here. I have added a terminal instruction for the same [here](https://github.com/kavinkuma6/cal.com/blob/add-cspell-pre-commit-to-fix-typos/.cspell/cspell.sh#L7) which would be shown in pre commit hook.

Incase, if we want to check the spelling for the whole project, unlike only the staged files as mentioned above, we can use the newly added script `npm run spell-check`



## Requirement/Documentation



- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change




## How should this be tested?



- Are there environment variables that should be set? - No
- What are the minimal test data to have? - NA
- What is expected (happy path) to have (input and output)? - NA
- Any other important info that could help to test that PR: I agree we have lot of variable name changes. But we can do most of the testing in the PR review process itself. Then we can use the other test setups to verify the overall workings. 


## Mandatory Tasks
